language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/toml/visitor/TomlSchemaVisitor.java

```diff
@@ -0,0 +1,206 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.commons.toml.visitor;
+
+import io.ballerina.toml.validator.schema.AbstractSchema;
+import io.ballerina.toml.validator.schema.ArraySchema;
+import io.ballerina.toml.validator.schema.BooleanSchema;
+import io.ballerina.toml.validator.schema.NumericSchema;
+import io.ballerina.toml.validator.schema.Schema;
+import io.ballerina.toml.validator.schema.SchemaVisitor;
+import io.ballerina.toml.validator.schema.StringSchema;
+import io.ballerina.toml.validator.schema.Type;
+import org.ballerinalang.langserver.commons.toml.common.TomlCommonUtil;
+import org.ballerinalang.langserver.commons.toml.common.TomlSyntaxTreeUtil;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.InsertTextFormat;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Schema Visitor for visiting toml validation schema.
+ *
+ * @since 2.0.0
+ */
+public class TomlSchemaVisitor extends SchemaVisitor {
+
+    private final Map<TomlNode, Map<String, CompletionItem>> completions = new HashMap<>();
+    private Map<String, CompletionItem> topLevelNodeStore;
+    private TomlNode topLevelNodeKey;
+
+    private Map<String, CompletionItem> initCompletionStore() {
+        if (this.topLevelNodeStore == null) {
+            return new HashMap<>();
+        }
+        return this.topLevelNodeStore;
+    }
```

Comment:
Is there a specific reason to have this method for initializing the `topLevelNodeStore`?

Reply:
Refactored the logic to be more clear